src/PlusServer/vtkPlusOpenIGTLinkServer.cxx

```diff
@@ -155,16 +160,24 @@ PlusStatus vtkPlusOpenIGTLinkServer::StartOpenIGTLinkService()
 
   if (this->ConnectionReceiverThreadId < 0)
   {
-    this->ConnectionActive.first = true;
+    this->ConnectionActive.Request = true;
     this->ConnectionReceiverThreadId = this->Threader->SpawnThread((vtkThreadFunctionType)&ConnectionReceiverThread, this);
   }
 
   if (this->DataSenderThreadId < 0)
   {
-    this->DataSenderActive.first = true;
+    this->DataSenderActive.Request = true;
     this->DataSenderThreadId = this->Threader->SpawnThread((vtkThreadFunctionType)&DataSenderThread, this);
   }
 
+  // Wait a short duration to see if both threads initialized properly
+  std::this_thread::sleep_for(std::chrono::milliseconds(250));

```

Comment:
self->ServerSocket->CreateServer(self->ListeningPort) is a networking task, so we cannot be sure that it completes within 250ms. The nicest would be to wait (with a timeout) for a mutex to be released, but retrying 10x-20x with some small delay between them would be a bit faster and more robust.

Reply:
What timeout would you recommend? I didn't want to delay server startup by very much, as this use case occurs very infrequently.

Reply:
You can use shorter timeouts (50ms) with multiple repeated checks (if not ready yet then wait again and re-check). Maybe up to 2 seconds in total.